pkg/component/watch_test.go

```diff
@@ -49,9 +58,266 @@ func TestIsRegExpMatch(t *testing.T) {
+				t.Errorf("expected %v, got %v", tt.want, matched)
 			}
 		})
 	}
 
 }
+
+// setUpF8AnalyticsComponentSrc sets up a mock analytics component source base for observing changes to source files.
+// Parameters:
+//	componentName: Name of the source directory
+//	requiredFilePaths: list of required sources, their description like whether regularfile/directory, parent directory path of source and desired modification type like update/create/delete/append
+// Returns:
+//	absolute base path of source code
+//	directory structure containing mappings from desired relative paths to their respective absolute path containing FileProperties.
+func setUpF8AnalyticsComponentSrc(componentName string, requiredFilePaths []testingutil.FileProperties) (string, map[string]testingutil.FileProperties, error) {
+	// retVal is mappings from desired relative paths to their respective absolute path containing FileProperties.
+	// This is required because ioutil#TempFile and ioutil#TempFolder creates paths with random numeric suffixes.
+	// So, to be able to refer to the file/folder at any later point in time the created paths returned by ioutil#TempFile or ioutil#TempFolder will need to be saved.
+	retVal := make(map[string]testingutil.FileProperties)
+	dirTreeMappings := make(map[string]string)
+	basePath := ""
+	// Create temporary directory for mock component source code
+	srcPath, err := testingutil.TempMkdir(basePath, componentName)
+	if err != nil {
+		return "", retVal, errors.Wrapf(err, "failed to create dir %s under %s", componentName, basePath)
+	}
+	dirTreeMappings[componentName] = srcPath
+
+	// For each of the passed(desired) files/folders under component source
+	for _, fileProperties := range requiredFilePaths {
+		// get relative path using file parent and file name passed
+		relativePath := filepath.Join(fileProperties.FileParent, fileProperties.FilePath)
+		// get its absolute path using the mappings preserved from previous creates
+		if realParentPath, ok := dirTreeMappings[fileProperties.FileParent]; ok {
+			// real path for the intended file operation is obtained from previously maintained directory tree mappings by joining parent path and file name
+			realPath := filepath.Join(realParentPath, fileProperties.FilePath)
+			// Preserve the new paths for further reference
+			fileProperties.FilePath = filepath.Base(realPath)
+			fileProperties.FileParent, _ = filepath.Rel(srcPath, filepath.Dir(realPath))
+		}
+		// Perform mock operation as requested by the parameter
+		newPath, err := testingutil.SimulateFileModifications(srcPath, fileProperties)
+		dirTreeMappings[relativePath] = newPath
+		if err != nil {
+			return "", retVal, errors.Wrapf(err, "unable to setup test env")
+		}
+		fileProperties.FilePath = filepath.Base(newPath)
+		fileProperties.FileParent = filepath.Dir(newPath)
+		retVal[relativePath] = fileProperties
+	}
+	// Return base source path and directory tree mappings
+	return srcPath, retVal, nil
```

Comment:
Please add some spacing / newlines in this paramter. I feel this is very clustered.

Reply:
@cdrage I think you meant spacing between the code lines right?
Done